### PR TITLE
codec: improve the performance of decoding decimal

### PR DESCRIPTION
**Pull request by zz-jason**

### CODE REVIEW

**Comment by winoros** [comment]
benchmark result?

**Comment by coocood** [comment]
LGTM

**Comment by zz-jason** [comment]
@winoros benchmark result:
before this patch:
```txt
➜ go test -bench BenchmarkDecodeDecimal -run xx -count 5
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/util/codec
BenchmarkDecodeDecimal-8        20000000                74.8 ns/op
BenchmarkDecodeDecimal-8        20000000                75.0 ns/op
BenchmarkDecodeDecimal-8        20000000                75.1 ns/op
BenchmarkDecodeDecimal-8        20000000                74.8 ns/op
BenchmarkDecodeDecimal-8        20000000                74.3 ns/op
PASS
ok      github.com/pingcap/tidb/util/codec      7.913s
```

after this patch:
```txt
➜ go test -bench BenchmarkDecodeDecimal -run xx -count 5
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/util/codec
BenchmarkDecodeDecimal-8        20000000                55.9 ns/op
BenchmarkDecodeDecimal-8        20000000                54.5 ns/op
BenchmarkDecodeDecimal-8        20000000                54.4 ns/op
BenchmarkDecodeDecimal-8        20000000                54.5 ns/op
BenchmarkDecodeDecimal-8        20000000                54.8 ns/op
PASS
ok      github.com/pingcap/tidb/util/codec      5.817s
```

performance gain is almost 38%

**Comment by zz-jason** [comment]
@shenli @winoros PTAL

File: expression/distsql_builtin.go [hunk 1]
```diff
@@ -684,10 +684,15 @@ func convertFloat(val []byte, f32 bool) (*Constant, error) {
 
 func convertDecimal(val []byte) (*Constant, error) {
 	_, dec, err := codec.DecodeDecimal(val)
+	var d types.Datum
```
Comment (ngaut) [inline expression/distsql_builtin.go]: DRY. We should abstract a function, it's duplicate with  util/codec/codec.go at line 339

**Pull request merged**